site/pages/admin/social-media-accounts.page.tsx

```diff
@@ -1,51 +1,180 @@
+import startCase from "lodash/startCase";
 import { NextPageContext } from "next";
 import Link from "next/link";
-import { Fragment, ReactElement } from "react";
+import { parseCookies } from "nookies";
+import { Fragment, ReactElement, ReactNode } from "react";
 import Table from "../../components/form/Table";
 import { BaseLayout } from "../../components/layout/Layout";
-import { SocialMediaAccountsSchema } from "../../schemas/social-media-accounts.schema";
+import { COOKIES_ID_TOKEN, COOKIES_REFRESH_TOKEN, HOOTSUITE_URL } from "../../constants";
+import { getParameter, getParametersByPath, putParameter } from "../../data/ssm";
+import { HootsuiteMe, HootsuiteSocialProfiles, HootsuiteToken } from "../../interfaces";
+import { HootsuiteProfiles, SocialMediaAccountsSchema } from "../../schemas/social-media-accounts.schema";
 import { toLowerStartCase } from "../../utils";
+import { getSessionWithOrgDetail } from "../../utils/apiUtils/auth";
 
 const title = "Social Media Accounts - Create Transport Disruptions Service";
 const description = "Social Media Accounts page for the Create Transport Disruptions Service";
 
+export const getHootsuiteData = async (
+    ctx: NextPageContext,
+    username: string,
+    orgId: string,
+): Promise<{ clientId: string; userData: SocialMediaAccountsSchema }> => {
+    const cookies = parseCookies(ctx);
+    let clientIdValue = "";
+    let userData: SocialMediaAccountsSchema = [];
+    try {
+        const idToken = cookies[COOKIES_ID_TOKEN];
+        const refreshToken = cookies[COOKIES_REFRESH_TOKEN];
+
+        const clientId = await getParameter(`/social/hootsuite/client_id`);
+        const clientSecret = await getParameter(`/social/hootsuite/client_secret`);
+        if (!clientId || !clientSecret) {
+            throw new Error("clientId and clientSecret must be defined");
+        }
+
+        clientIdValue = clientId.Parameter?.Value || "";
+        const key = `${clientId.Parameter?.Value || ""}:${clientSecret.Parameter?.Value || ""}`;
+
+        const authToken = `Basic ${Buffer.from(key).toString("base64")}`;
+
+        if (idToken && refreshToken)
+            await Promise.all([
+                putParameter(`/${username}/token`, idToken, "SecureString", true),
+                putParameter(`/${username}/refresh-token`, refreshToken, "SecureString", true),
+            ]);
+
+        const tokensByOrganisation = await getParametersByPath(`/social/${orgId}/hootsuite`);
```

Comment:
Same for this api too. All these can be invoked simultaneously using Promise.all

Reply:
done

Reply:
dont think this is done as I am unable to find this change